internal/cluster/cluster_test.go

```diff
@@ -2836,24 +2841,26 @@ var _ = Describe("Deregister inactive clusters", func() {
 	})
 
 	It("Deregister inactive cluster limited", func() {
-		inactiveCluster1 := registerCluster()
-		inactiveCluster2 := registerCluster()
-		inactiveCluster3 := registerCluster()
-		inactiveCluster4 := registerCluster()
-		inactiveCluster5 := registerCluster()
-		inactiveCluster6 := registerCluster()
+		var (
+			clustersBeforeDeregisterInactive []*common.Cluster
+			clustersAfterDeregisterInactive  []*common.Cluster
+		)
+
+		for i := 0; i < 6; i++ {
+			registerCluster()
+		}
 
 		lastActive := strfmt.DateTime(time.Now())
 
-		Expect(state.DeregisterInactiveCluster(ctx, 3, lastActive)).ShouldNot(HaveOccurred())
+		err := db.Find(&clustersBeforeDeregisterInactive).Error
+		Expect(err).To(BeNil())
+		Expect(clustersBeforeDeregisterInactive).To(HaveLen(7))
 
-		Expect(wasDeregisterd(db, *inactiveCluster1.ID)).To(BeTrue())
-		Expect(wasDeregisterd(db, *inactiveCluster2.ID)).To(BeTrue())
+		Expect(state.DeregisterInactiveCluster(ctx, 3, lastActive)).ShouldNot(HaveOccurred())
 
-		Expect(wasDeregisterd(db, *inactiveCluster3.ID)).To(BeFalse())
-		Expect(wasDeregisterd(db, *inactiveCluster4.ID)).To(BeFalse())
-		Expect(wasDeregisterd(db, *inactiveCluster5.ID)).To(BeFalse())
-		Expect(wasDeregisterd(db, *inactiveCluster6.ID)).To(BeFalse())
+		err = db.Find(&clustersAfterDeregisterInactive).Error
+		Expect(err).To(BeNil())
```

Comment:
I guess both are valid:
```sh
git grep "(err).ToNot(HaveOccurred())" |  wc -l
     624
git grep "(err).To(BeNil())" |  wc -l
     397
```

I'll be leaving it as is.

Reply:
I just chose to leave it as is since there are both the same. No real agenda behind it.
But per your comment, I changed it to `Expect(err).ToNot(HaveOccurred())`

Reply:
Thanks!